tests/integration/test_local_client.py

```diff
@@ -0,0 +1,279 @@
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+import pytest
+
+import re
+
+import numpy as np
+import pandas as pd
+
+from antares import create_study_local
+from antares.exceptions.exceptions import (
+    AreaCreationError,
+    InvalidNameError,
+    LinkCreationError,
+)
+from antares.model.area import AdequacyPatchMode, Area, AreaProperties, AreaUi
+from antares.model.binding_constraint import BindingConstraintProperties, ClusterData, ConstraintTerm, LinkData
+from antares.model.commons import FilterOption
+from antares.model.link import Link, LinkProperties, LinkUi
+from antares.model.renewable import RenewableClusterGroup, RenewableClusterProperties
+from antares.model.settings.general import GeneralParametersLocal, Mode
+from antares.model.settings.playlist_parameters import PlaylistParameters
+from antares.model.settings.study_settings import StudySettingsLocal
+from antares.model.st_storage import STStorageGroup, STStorageProperties
+from antares.model.study import Study
+from antares.model.thermal import ThermalCluster, ThermalClusterGroup, ThermalClusterProperties
+from antares.tools.ini_tool import IniFile, IniFileTypes
+from antares.tools.time_series_tool import TimeSeriesFileType
+
+
+class TestLocalClient:
+    def test_local_study(self, tmp_path, other_area):
+        study_name = "test study"
+        study_version = "880"
+
+        # Study
+        test_study = create_study_local(study_name, study_version, tmp_path.absolute())
+        assert isinstance(test_study, Study)
+
+        # Areas
+        fr = test_study.create_area("fr")
+        at = test_study.create_area("at")
+
+        assert isinstance(fr, Area)
+        assert isinstance(at, Area)
+
+        ## Area already exists
+        with pytest.raises(
+            AreaCreationError,
+            match="Could not create the area fr: There is already an area 'fr' in the study 'test study'",
+        ):
+            test_study.create_area("fr")
+
+        # Link
+        at_fr = test_study.create_link(area_from=fr.id, area_to=at.id)
+
+        assert isinstance(at_fr, Link)
+
+        ## Cannot link areas that don't exist in the study
+        with pytest.raises(LinkCreationError, match="Could not create the link fr / usa: usa does not exist"):
+            test_study.create_link(area_from=fr.id, area_to=other_area.id)
+
+        # Thermal
+        fr_nuclear = fr.create_thermal_cluster("nuclear")
+
+        assert isinstance(fr_nuclear, ThermalCluster)
+
+        # Setup time series for following tests
+        time_series_rows = 10  # 365 * 24
+        time_series_columns = 1
+        time_series = pd.DataFrame(np.around(np.random.rand(time_series_rows, time_series_columns)))
+
+        # Load
+        fr.create_load(time_series)
+
+        assert test_study.service.config.study_path.joinpath(
```

Comment:
You should test with the real path not the TimeSeriesFileType as if i write a wrong path inside the object the test will still pass and yet it will be wrong for the simulator

Same for the rest of the test

Reply:
Updated